.githooks/commit-msg

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+## the first arg is the path to the git commit temporary file
+TEMPORARY_FILE_PATH=$1
+
+## get commit-message from the temporary file
+COMMIT_MSG=`head -n1 $TEMPORARY_FILE_PATH`
+
+## init regex to match commit-message format
+REGEX="#[0-9]+ \w+"
```

Comment:
`REGEX` or `COMMIT_MSG_REGEX`?

Reply:
instead of `#issueNumber My commit message` can you impose `My commit message #issueNumber` or `I-#issueNumber My commit message`

Won't it clash if someone tries to rebase some commits? During rebase sentences starting with # are skipped as comments.

any thoughts?

Reply:
That is an excellent point. Merge commits had skipped my mind. Modified the regex to consider default merge commit message in commit [da585e6](https://github.com/CraftsmenLtd/BloodConnect/pull/18/commits/da585e6146c32534b05a2fb3f3459da9a388eb78). If the default commit is changed manually, I believe the issue number can be put also. Does this resolve the issue or are there any other concerns I am missing?